Frameworks/QuartzCore/CALayer.mm

```diff
@@ -833,6 +831,8 @@ - (void)removeFromSuperlayer {
     [CATransaction _removeLayer:self];
 
     pSuper->priv->removeChild(self);
+
+    [pSuper setNeedsLayout];
     [self release];
```

Comment:
>[self release]; [](start = 4, length = 15)

wow - that's kind of odd.  you'd think that the superlayer would be holding the reference....?